### PR TITLE
Feature/170

### LINKED ISSUE #170: Use OpenShift OAuth for SonarQube

### PR DESCRIPTION
**Pull request by ungerts**
Closes #170 

Actual Todos:
* Align with example config
* Update documentation

### CODE REVIEW

File: sonarqube/run.sh [hunk 1]
```diff
@@ -13,6 +13,11 @@ if [ $SONAR_AUTH_CROWD == true ]; then
   echo "crowd.application=$SONARQUBE_CROWD_APP" >> conf/sonar.properties
   echo "crowd.password=$SONARQUBE_CROWD_PWD" >> conf/sonar.properties
   echo "sonar.security.localUsers=admin" >> conf/sonar.properties
+elif [ $SONAR_OAUTH_PLUGIN_ENABLED == true ]; then
+  echo "sonar.auth.openshift.isEnabled=true" >> conf/sonar.properties
+  echo "sonar.auth.openshift.sar.groups=$SONAR_OAUTH_ADMIN_GROUP=sonar-administrators,$SONAR_OAUTH_USERS_GROUP=sonar-users" >> conf/sonar.properties
+  echo "oauth.cert=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt" >> conf/sonar.properties
+  echo 'kubernetes.service=https://${env:KUBERNETES_SERVICE_HOST}:${env:KUBERNETES_SERVICE_PORT}/' >> conf/sonar.properties
```
Comment (michaelsauter) [inline sonarqube/run.sh]: Could you also explain the effects of this? What does that mean for users, admins, ... ?

File: sonarqube/ocp-config/sonarqube.yml [hunk 2]
```diff
@@ -382,40 +405,46 @@ objects:
...
     type: ClusterIP
 parameters:
+- displayName: OC project
+  name: OC_PROJECT
+  description: Name of the ocp project to deploy to. This is normally the "cd" project
+  required: true
+# this must be base64 encoded.
 - displayName: Admin Password
-  name: SONAR_ADMIN_PASSWORD_B64
+  name: ADMIN_PASSWORD
```
Comment (michaelsauter) [inline sonarqube/ocp-config/sonarqube.yml]: Hmm, you are reverting my changes with this. Please see #93.

**Comment by michaelsauter** [comment]
@ungerts @tjaeschke After merging #250, I will also open a PR into this one to adjust it to the new SonarQube structure. You do not need to do that.

File: sonarqube/ocp-config/sonarqube.yml [hunk 3]
```diff
@@ -3,6 +3,21 @@ kind: Template
 metadata:
   name: sonarqube
 objects:
+- apiVersion: v1
+  kind: ServiceAccount
+  metadata:
```
Comment (michaelsauter) [inline sonarqube/ocp-config/sonarqube.yml]: Missing label app=sonarqube

File: sonarqube/ocp-config/sonarqube.yml [hunk 4]
```diff
@@ -312,9 +339,9 @@ objects:
     wildcardPolicy: None
 - apiVersion: v1
   data:
-    admin-password: ${SONAR_ADMIN_PASSWORD_B64}
-    auth-token: ${SONAR_AUTH_TOKEN_B64}
-    crowd-password: ${SONAR_CROWD_PASSWORD_B64}
+    admin-password: ${ADMIN_PASSWORD}
```
Comment (kiwo) [inline sonarqube/ocp-config/sonarqube.yml]:
```suggestion
    admin-password: ${SONAR_ADMIN_PASSWORD_B64}
```

File: sonarqube/ocp-config/sonarqube.yml [hunk 5]
```diff
@@ -312,9 +339,9 @@ objects:
...
+    auth-token: ${AUTH_TOKEN}
```
Comment (kiwo) [inline sonarqube/ocp-config/sonarqube.yml]:
```suggestion
    auth-token: ${SONAR_AUTH_TOKEN_B64}
```

File: sonarqube/ocp-config/sonarqube.yml [hunk 6]
```diff
@@ -312,9 +339,9 @@ objects:
...
+    crowd-password: ${CROWD_PASSWORD}
```
Comment (kiwo) [inline sonarqube/ocp-config/sonarqube.yml]:
```suggestion
    crowd-password: ${SONAR_CROWD_PASSWORD_B64}
```

File: sonarqube/ocp-config/sonarqube.yml [hunk 7]
```diff
@@ -328,15 +355,15 @@ objects:
...
+    crowd-application: ${CROWD_APPLICATION}
     crowd-url: ${CROWD_URL}
     database-jdbc-url: ${SONAR_DATABASE_JDBC_URL}
     database-name: ${SONAR_DATABASE_NAME}
     database-user: ${SONAR_DATABASE_USER}
     sonarqube-admin-user: ${SONAR_ADMIN_USERNAME}
 - apiVersion: v1
   data:
-    database-password: ${SONAR_DATABASE_PASSWORD_B64}
+    database-password: ${DATABASE_PASSWORD}
```
Comment (kiwo) [inline sonarqube/ocp-config/sonarqube.yml]:
```suggestion
    database-password: ${SONAR_DATABASE_PASSWORD_B64}
```

File: sonarqube/ocp-config/sonarqube.yml [hunk 8]
```diff
@@ -328,15 +355,15 @@ objects:
     name: sonarqube
   apiVersion: v1
   data:
-    crowd-application: ${SONAR_CROWD_APPLICATION}
+    crowd-application: ${CROWD_APPLICATION}
```
Comment (kiwo) [inline sonarqube/ocp-config/sonarqube.yml]:
```suggestion
    crowd-application: ${SONAR_CROWD_APPLICATION}
```

File: sonarqube/ocp-config/sonarqube.yml [hunk 9]
```diff
@@ -391,39 +418,40 @@ parameters:
   description: Git ref to use for source code.
   required: true
 - displayName: Admin Password
-  name: SONAR_ADMIN_PASSWORD_B64
+  name: ADMIN_PASSWORD
   description: The base64 encoded admin password
   required: true
 - displayName: Admin Username
-  name: SONAR_ADMIN_USERNAME
+  name: ADMIN_USERNAME
```
Comment (kiwo) [inline sonarqube/ocp-config/sonarqube.yml]:
```suggestion
  name: SONAR_ADMIN_USERNAME
```

File: sonarqube/ocp-config/sonarqube.yml [hunk 10]
```diff
@@ -391,39 +418,40 @@ parameters:
...
   description: the sonarqube admin username
   required: true
 - displayName: Auth Token
-  name: SONAR_AUTH_TOKEN_B64
+  name: AUTH_TOKEN
```
Comment (kiwo) [inline sonarqube/ocp-config/sonarqube.yml]:
```suggestion
  name: SONAR_AUTH_TOKEN_B64
```

File: sonarqube/ocp-config/sonarqube.yml [hunk 11]
```diff
@@ -391,39 +418,40 @@ parameters:
...
   required: true
 - displayName: Crowd Application
-  name: SONAR_CROWD_APPLICATION
+  name: CROWD_APPLICATION
```
Comment (kiwo) [inline sonarqube/ocp-config/sonarqube.yml]:
```suggestion
  name: SONAR_CROWD_APPLICATION
```

File: sonarqube/ocp-config/sonarqube.yml [hunk 12]
```diff
@@ -391,39 +418,40 @@ parameters:
...
   description: application name as setup in crowd for authn/z
 - displayName: Crowd Password
-  name: SONAR_CROWD_PASSWORD_B64
+  name: CROWD_PASSWORD
```
Comment (kiwo) [inline sonarqube/ocp-config/sonarqube.yml]:
```suggestion
  name: SONAR_CROWD_PASSWORD_B64
```

File: sonarqube/ocp-config/sonarqube.yml [hunk 13]
```diff
@@ -391,39 +418,40 @@ parameters:
...
   description: base64 encoded application password as setup in crowd for authn/z
 - displayName: Crowd URL
   name: CROWD_URL
   description: url under which crowd is reachable from sonaruqbe
+# e.g. jdbc:postgresql://sonarqube-postgresql:5432/sonarqube
 - displayName: Database JDBC URL
-  name: SONAR_DATABASE_JDBC_URL
-  description: database jdbc url, e.g. jdbc:postgresql://sonarqube-postgresql:5432/sonarqube
+  name: DATABASE_JDBC_URL
```
Comment (kiwo) [inline sonarqube/ocp-config/sonarqube.yml]:
```suggestion
  name: SONAR_DATABASE_JDBC_URL
```

File: sonarqube/ocp-config/sonarqube.yml [hunk 14]
```diff
@@ -391,39 +418,40 @@ parameters:
   description: Git ref to use for source code.
   required: true
 - displayName: Admin Password
-  name: SONAR_ADMIN_PASSWORD_B64
+  name: ADMIN_PASSWORD
```
Comment (kiwo) [inline sonarqube/ocp-config/sonarqube.yml]:
```suggestion
  name: SONAR_ADMIN_PASSWORD_B64
```

File: sonarqube/ocp-config/sonarqube.yml [hunk 15]
```diff
@@ -391,39 +418,40 @@ parameters:
...
+# e.g. jdbc:postgresql://sonarqube-postgresql:5432/sonarqube
 - displayName: Database JDBC URL
-  name: SONAR_DATABASE_JDBC_URL
-  description: database jdbc url, e.g. jdbc:postgresql://sonarqube-postgresql:5432/sonarqube
+  name: DATABASE_JDBC_URL
+  description: database jdbc url
   required: true
 - displayName: Database Name
-  name: SONAR_DATABASE_NAME
+  name: DATABASE_NAME
```
Comment (kiwo) [inline sonarqube/ocp-config/sonarqube.yml]:
```suggestion
  name: SONAR_DATABASE_NAME
```

File: sonarqube/ocp-config/sonarqube.yml [hunk 16]
```diff
@@ -391,39 +418,40 @@ parameters:
...
   description: database name, e.g. sonarqube
   required: true
 - displayName: Database Password
-  name: SONAR_DATABASE_PASSWORD_B64
+  name: DATABASE_PASSWORD
```
Comment (kiwo) [inline sonarqube/ocp-config/sonarqube.yml]:
```suggestion
  name: SONAR_DATABASE_PASSWORD_B64
```

File: sonarqube/ocp-config/sonarqube.yml [hunk 17]
```diff
@@ -391,39 +418,40 @@ parameters:
...
   description: base64 encoded database user password
   required: true
 - displayName: Database User
-  name: SONAR_DATABASE_USER
+  name: DATABASE_USER
```
Comment (kiwo) [inline sonarqube/ocp-config/sonarqube.yml]:
```suggestion
  name: SONAR_DATABASE_USER
```

**Pull request closed without merging**